src/select/select.js

```diff
@@ -350,6 +359,40 @@ class Select extends React.Component<PropsT, StatelessStateT> {
     const {type, multiple} = this.props;
     return type === TYPE.search ? true : multiple;
   }
+
+  handledHotKeys(
```

Comment:
it seems that this function has side-effects - should we split it into two functions, one that translates `e.key` into a state like `isDropDownOpen: true`, and one that actually sets it?

Reply:
no side effects. Function does one single action and notifies outside if this action has happened